### ISSUE TITLE
Arguments spread doesn't take into account actual array length

### ISSUE THREAD

**Issue by sebek64** [Microsoft/TypeScript#19419]
**TypeScript Version:**  2.6.0-rc, 2.7.0-dev.20171021

**Code**

```ts
function a(first: string, ...args: string[]);
function a(...args: string[]) {
}

const b = ["a", "x"];

if (a.length > 0) {
    a(...b);
}
```

**Expected behavior:**
Should not produce error.

**Actual behavior:**\
Produces
error TS2557: Expected at least 1 arguments, but got a minimum of 0.

**Comment by Ghabriel** [comment]
I created an issue yesterday that seems to be about the same problem: #19403 
No answers yet, though. Nice to know that I'm not the only one who encountered this problem.

**Comment by sebek64** [comment]
@Ghabriel oops, I checked for duplicities a few days ago and found nothing. Today, I just reported the problem.

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/4130

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
